doc/source/ray-observability/key-concepts.rst

```diff
@@ -89,54 +66,45 @@ drop into a PDB session that you can then use to:
 - Step within that task or actor
 - Move up or down the stack
 
-See :ref:`Ray Debugger <ray-debugger>` for more details.
+View :ref:`Ray Debugger <ray-debugger>` for more details.
+
+.. _profiling-concept:
 
 Profiling
 ---------
-Ray is compatible with Python profiling tools such as ``CProfile``. It also supports its built-in profiling tool such as :ref:`ray timeline <ray-timeline-doc>`.
+Profiling is way of analyzing the performance of an application by sampling the resource usage of it. Ray supports various profiling tools:
+
+- CPU profiling for worker processes, including integration with :ref:`py-spy <dashboard-profiling>` and :ref:`cProfile <dashboard-cprofile>`
+- Memory profiling for worker processes with :ref:`memray <ray-core-mem-profiling>`
+- Built in task/actor profiling tool called :ref:`ray timeline <ray-core-timeline>`
 
-See :ref:`Profiling <dashboard-cprofile>` for more details.
+Ray doesn't provide native integration with GPU profiling tools. Try running GPU profilers like PyTorch Profiler without Ray to identify the issues.
```

Comment:
Add a link to pytorch profiler? 

I think we can technically use runtime env to do this, but I can make changes in the future